packages/block-editor/src/store/effects.js

```diff
@@ -116,7 +116,7 @@ export default {
 				{
 					...blockA,
 					attributes: {
-						...blockA.attributes,
+						...( blockA.attributes.content.length === 0 ? blockB.attributes : blockA.attributes ),
```

Comment:
`mergeBlocks` is not specific to paragraph. We can't assume paragraph attributes here.

I wasn't really sure where this original logic came from. I found the earliest implementation from long ago in #461 . I think the general idea is that if there's two blocks of the same type, then their attributes should be merged together.

Which makes me think then that it might be correct to say that a correct behavior to apply the attributes of `blockB` _into_ `blockA` is:

```suggestion
						...blockA.attributes,
						...blockB.attributes,
```